data-pipeline/src/agent_info/fetcher.rs

```diff
@@ -335,12 +335,14 @@ mod tests {
             fetcher.run().await;
         });
 
-        // Wait for first fetch
-        while mock_v1.hits_async().await == 0 {
+        // Wait until the info is fetched
+        while info.load().is_none() {
```

Comment:
Can we have a timeout mechanism here? It can be something generous, but we probably don't want to hang for minutes here.